WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationDomainAdapter.java

```diff
@@ -0,0 +1,199 @@
+        mOnAdapterListener = onAdapterListener;
+    }
+
+    public void setData(boolean isLoading, String keywords, List<DomainSuggestionResponse> suggestions) {
+        if (isLoading != mIsLoading) {
+            notifyItemChanged(1);
+        }
+
+        mIsLoading = isLoading;
+        mKeywords = keywords;
+        mSuggestions = suggestions;
+
+        if (mSuggestions != null) {
+            mSelectedDomain = mSuggestions.get(0);
+            mOnAdapterListener.onSelectionChange(mSelectedDomain.domain_name);
+        }
+
+        notifyDataSetChanged();
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        if (viewType == VIEW_TYPE_HEADER) {
+            View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.site_creation_domain_header,
+                    parent, false);
+            return new HeaderViewHolder(itemView);
+        } else if (viewType == VIEW_TYPE_INPUT) {
+            View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.site_creation_domain_input, parent,
+                    false);
+            return new InputViewHolder(itemView);
+        } else {
+            View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.site_creation_domain_item, parent,
+                    false);
+            return new DomainViewHolder(itemView);
+        }
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        int viewType = getItemViewType(position);
+
+        if (viewType == VIEW_TYPE_HEADER) {
+//            final HeaderViewHolder headerViewHolder = (HeaderViewHolder) holder;
+        } else if (viewType == VIEW_TYPE_INPUT) {
+            final InputViewHolder inputViewHolder = (InputViewHolder) holder;
+            inputViewHolder.progressBar.setVisibility(mIsLoading ? View.VISIBLE : View.GONE);
+            if (inputViewHolder.input.getTag() == null) {
+                inputViewHolder.input.setText(mKeywords);
+                inputViewHolder.input.setTag(Boolean.TRUE);
+                inputViewHolder.input.addTextChangedListener(new TextWatcher() {
+                    @Override
+                    public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                    }
+
+                    @Override
+                    public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                    }
+
+                    @Override
+                    public void afterTextChanged(final Editable editable) {
+                        mDebouncer.debounce(Void.class, new Runnable() {
+                            @Override
+                            public void run() {
+                                mOnAdapterListener.onKeywordsChange(editable.toString());
+                            }
+                        }, 400, TimeUnit.MILLISECONDS);
```

Comment:
We should make this a constant to help us use the same value for similar interfaces across the app.  As an example, [the Username Changer screen uses `2000` milliseconds](https://github.com/wordpress-mobile/WordPress-Android/pull/7151/files#diff-4e6863adecd7157709266b8ea7819618R72).

Reply:
Done with 7741abc.